### PR TITLE
feat(orchestrator-form-widgets): add ActiveBoolean widget for dynamic boolean form fields

### PR DESCRIPTION
**Pull request by lokanandaprabhu**
## Summary

Adds a new **ActiveBoolean** widget that enables boolean (checkbox) fields in Orchestrator workflow forms to dynamically fetch their values from external APIs, following the same ui:props pattern as existing Active widgets (ActiveTextInput, ActiveDropdown, ActiveMultiSelect).

## Changes

- ✅ New `ActiveBoolean` widget component
- ✅ Comprehensive unit tests
- ✅ Documentation in `orchestratorFormWidgets.md`
- ✅ Widget registration in `FormDecoratorContent.tsx`
- ✅ Changeset for release notes

## Features

- Supports `fetch:url` to call external APIs
- Supports `fetch:response:value` JSONata selector to extract boolean values
- Supports `fetch:response:default` as static fallback
- Coerces string values (`"true"`, `"false"`, `"1"`, `"0"`) and numbers (`1`, `0`) to boolean
- Supports `fetch:retrigger` to refetch when other fields change
- Supports `fetch:clearOnRetrigger` to clear value before refetching
- Supports all standard fetch configuration: retry, error handling, etc.
- Tracks user changes to prevent overwriting manual edits
- Shows loading spinner while fetching
- Respects `readOnly` schema property

## Video:

https://github.com/user-attachments/assets/73637f87-dc43-4f5d-b6b2-1eac151ca95f

## Usage Example

```json
{
  "enableFeature": {
    "type": "boolean",
    "title": "Enable Advanced Features",
    "ui:widget": "ActiveBoolean",
    "ui:props": {
      "fetch:url": "https://api.example.com/feature-flags",
      "fetch:response:value": "features.advanced.enabled",
      "fetch:response:default": false,
      "fetch:retrigger": ["current.tenantId"]
    }
  }
}
```

---

### 1. Setup Proxy Configuration

Add this to your `app-config.local.yaml`:

```yaml
proxy:
  endpoints:
    "/dummyjson":
      target: "https://dummyjson.com"
      changeOrigin: true
      allowedMethods: ["GET"]
      allowedHeaders: ["Content-Type"]
      headers:
        X-Requested-With: "XMLHttpRequest"
```

### 2. Create Test Workflow

Create the following files in your dev environment:

**File: `packages/backend/.devModeTemp/repository/workflows/test-active-boolean.sw.yaml`**

```yaml
id: test-active-boolean
version: '1.0'
specVersion: '0.8'
name: Test ActiveBoolean Widget
description: Testing ActiveBoolean widget with various fetch and ui:props configurations
annotations:
  - "workflow-type/infrastructure"
dataInputSchema: schemas/test-active-boolean__main-schema.json
extensions:
  - extensionid: workflow-output-schema
    outputSchema: schemas/workflow-output-schema.json
start: StartState
functions:
  - name: runActionFunction
    type: custom
    operation: sysout
  - name: successResult
    type: expression
    operation: '{ "result": { "message": "ActiveBoolean test completed successfully" } }'
states:
  - name: StartState
    type: operation
    actions:
      - name: runAction
        functionRef:
          refName: runActionFunction
          arguments:
            message: '"Workflow completed with input: " + (. | tostring)'
      - name: setOutput
        functionRef:
          refName: successResult
    end: true
```

**File: `packages/backend/.devModeTemp/repository/workflows/schemas/test-active-boolean__main-schema.json`**

<details>
<summary>Click to expand schema (169 lines)</summary>

```json
{
  "$id": "classpath:/schemas/test-active-boolean__main-schema.json",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "Test ActiveBoolean Widget",
  "type": "object",
  "description": "Testing ActiveBoolean widget with various fetch and ui:props configurations",
  "properties": {
    "basicTests": {
      "type": "object",
      "title": "Basic ActiveBoolean Tests",
      "description": "Test basic fetch capabilities of ActiveBoolean widget",
      "properties": {
        "basicFeature": {
          "type": "boolean",
          "title": "Basic Feature Flag",
          "description": "Fetches boolean value from API - uses stock > 50 as boolean",
          "ui:widget": "ActiveBoolean",
          "ui:props": {
            "fetch:url": "$${{backend.baseUrl}}/api/proxy/dummyjson/products/1",
            "fetch:response:value": "stock > 50"
          }
        },
        "featureWithDefault": {
          "type": "boolean",
          "title": "Feature with Static Default",
          "description": "Has a static default that shows while fetch is loading",
          "ui:widget": "ActiveBoolean",
          "ui:props": {
            "fetch:url": "$${{backend.baseUrl}}/api/proxy/dummyjson/products/2",
            "fetch:response:value": "stock > 100",
            "fetch:response:default": true
          }
        },
        "stringCoercion": {
          "type": "boolean",
          "title": "String to Boolean Coercion",
          "description": "API returns string, widget coerces to boolean",
          "ui:widget": "ActiveBoolean",
          "ui:props": {
            "fetch:url": "$${{backend.baseUrl}}/api/proxy/dummyjson/products/3",
            "fetch:response:value": "availabilityStatus != 'Low Stock'",
            "fetch:response:default": false
          }
        },
        "numberCoercion": {
          "type": "boolean",
          "title": "Number to Boolean Coercion",
          "description": "API returns number, widget coerces to boolean",
          "ui:widget": "ActiveBoolean",
          "ui:props": {
            "fetch:url": "$${{backend.baseUrl}}/api/proxy/dummyjson/products/4",
            "fetch:response:value": "stock",
            "fetch:response:default": false
          }
        }
      }
    },
    "retriggerTests": {
      "type": "object",
      "title": "Retrigger and Dependency Tests",
      "description": "Test fetch:retrigger and dependent boolean fields",
      "properties": {
        "tenantId": {
          "type": "string",
          "title": "Product ID (Simulating Tenant)",
          "description": "Select a product ID to see different boolean values",
          "enum": ["5", "10", "15"],
          "default": "5"
        },
        "dependentFeature": {
          "type": "boolean",
          "title": "Tenant-Specific Feature",
          "description": "This boolean refetches when Product ID changes",
          "ui:widget": "ActiveBoolean",
          "ui:props": {
            "fetch:url": "$${{backend.baseUrl}}/api/proxy/dummyjson/products/$${{current.retriggerTests.tenantId}}",
            "fetch:response:value": "stock > 30",
            "fetch:retrigger": ["current.retriggerTests.tenantId"]
          }
        },
        "clearOnRetriggerFeature": {
          "type": "boolean",
          "title": "Feature with Clear on Retrigger",
          "description": "Clears before refetching",
          "ui:widget": "ActiveBoolean",
          "ui:props": {
            "fetch:url": "$${{backend.baseUrl}}/api/proxy/dummyjson/products/$${{current.retriggerTests.tenantId}}",
            "fetch:response:value": "price > 500",
            "fetch:retrigger": ["current.retriggerTests.tenantId"],
            "fetch:clearOnRetrigger": true
          }
        }
      }
    },
    "errorHandlingTests": {
      "type": "object",
      "title": "Error Handling Tests",
      "description": "Test error handling and retry behavior",
      "properties": {
        "silentErrorFeature": {
          "type": "boolean",
          "title": "Silent Error Handling",
          "description": "Errors are suppressed, shows default value",
          "ui:widget": "ActiveBoolean",
          "ui:props": {
            "fetch:url": "$${{backend.baseUrl}}/api/proxy/dummyjson/products/999999",
            "fetch:response:value": "stock > 0",
            "fetch:response:default": false,
            "fetch:error:silent": true
          }
        },
        "retryFeature": {
          "type": "boolean",
          "title": "Feature with Retry",
          "description": "Retries up to 3 times with exponential backoff",
          "ui:widget": "ActiveBoolean",
          "ui:props": {
            "fetch:url": "$${{backend.baseUrl}}/api/proxy/dummyjson/products/6",
            "fetch:response:value": "stock > 0",
            "fetch:retry:maxAttempts": 3,
            "fetch:retry:delay": 1000,
            "fetch:retry:backoff": 2,
            "fetch:retry:statusCodes": [404, 408, 429, 500, 502, 503, 504]
          }
        }
      }
    },
    "advancedTests": {
      "type": "object",
      "title": "Advanced Tests",
      "description": "Advanced scenarios and edge cases",
      "properties": {
        "readOnlyFeature": {
          "type": "boolean",
          "title": "Read-Only Feature Flag",
          "description": "This boolean is fetched but cannot be changed",
          "readOnly": true,
          "ui:widget": "ActiveBoolean",
          "ui:props": {
            "fetch:url": "$${{backend.baseUrl}}/api/proxy/dummyjson/products/7",
            "fetch:response:value": "stock > 0"
          }
        },
        "skipInitialFeature": {
          "type": "boolean",
          "title": "Skip Initial Value",
          "description": "Doesn't apply fetched value initially",
          "ui:widget": "ActiveBoolean",
          "ui:props": {
            "fetch:url": "$${{backend.baseUrl}}/api/proxy/dummyjson/products/1",
            "fetch:response:value": "stock > 50",
            "fetch:skipInitialValue": true
          }
        },
        "complexSelectorFeature": {
          "type": "boolean",
          "title": "Complex JSONata Selector",
          "description": "Uses complex JSONata expression",
          "ui:widget": "ActiveBoolean",
          "ui:props": {
            "fetch:url": "$${{backend.baseUrl}}/api/proxy/dummyjson/products/8",
            "fetch:response:value": "(price > 100) and (stock > 10)",
            "fetch:response:default": false
          }
        }
      }
    }
  }
}
```

</details>

---

## Supported ui:props

All standard Active widget props are supported:

- `fetch:url` - API endpoint
- `fetch:response:value` - JSONata selector for boolean value
- `fetch:response:default` - Static fallback value
- `fetch:retrigger` - Array of form field paths to watch
- `fetch:clearOnRetrigger` - Clear value before refetching
- `fetch:skipInitialValue` - Don't apply initial fetch result
- `fetch:retry:maxAttempts` - Number of retry attempts
- `fetch:retry:delay` - Initial retry delay (ms)
- `fetch:retry:backoff` - Backoff multiplier
- `fetch:retry:statusCodes` - HTTP status codes to retry
- `fetch:error:silent` - Suppress error messages
- `fetch:method`, `fetch:headers`, `fetch:body` - Request configuration

### CODE REVIEW

File: workspaces/orchestrator/plugins/orchestrator-form-widgets/src/widgets/ActiveBoolean.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+  const uiProps = useMemo(
+    () => (props.options?.props ?? {}) as UiProps,
+    [props.options?.props],
+  );
+  const isReadOnly = !!props?.schema.readOnly;
+
+  const defaultValueSelector = uiProps['fetch:response:value']?.toString();
+  const staticDefault = uiProps['fetch:response:default'];
+  const hasStaticDefault = staticDefault !== undefined;
```
Comment (karthikjeeyar) [inline workspaces/orchestrator/plugins/orchestrator-form-widgets/src/widgets/ActiveBoolean.tsx]:
```suggestion
  const staticDefault = coerceToBoolean(uiProps['fetch:response:default']);
  const hasStaticDefault = staticDefault !== undefined;
```

Default value should be also coerced like `fetch:response:value`, if not adding `fetch:response:default: "true"` will not be checked.

Comment (karthikjeeyar) [reply workspaces/orchestrator/plugins/orchestrator-form-widgets/src/widgets/ActiveBoolean.tsx]: If possible, add test some tests for this scenario.

**Review by karthikjeeyar — COMMENTED** [review]
Overall looks good, just one comment

**Review by lokanandaprabhu — COMMENTED** [review]
✅ Fixed! Applied your suggestion to coerce `fetch:response:default` values to boolean.

Now string defaults like `"true"`, `"false"`, `"1"`, `"0"` will be properly coerced using the same `coerceToBoolean` function.

Also added a test case to verify string default coercion works correctly.

Commit: 8d61aff

**Review by karthikjeeyar — APPROVED** [review]
/approve
/lgtm
